agent/osv_output_handler.py

```diff
@@ -79,48 +87,87 @@ def parse_results(output: str, api_key: str | None = None) -> Iterator[Vulnerabi
-            for vuln in package.get("vulnerabilities", []):
-                cve_ids = vuln.get("aliases", "")
-                if cve_ids == "" and vuln.get("references") is not None:
-                    cve_ids = _extract_cve_reference_advisory(vuln["references"])
-                risk_rating, cve_list_details = _aggregate_cves(
-                    cve_ids=cve_ids, api_key=api_key
-                )
-                description = (
-                    f"Dependency `{package_name}` with version `{package_version}`"
-                    f" found in the `{file_type}` `{file_name}` "
-                    f"has a security issue.\nThe issue is identified by CVEs: `{', '.join(cve_ids)}`."
-                )
-                recommendation = f"We recommend updating `{package_name}` to the latest available version."
-                yield Vulnerability(
-                    entry=kb.Entry(
-                        title=f"Use of Outdated Vulnerable Component: "
-                        f"{package_name}@{package_version}: {', '.join(cve_ids)}",
-                        risk_rating=risk_rating.upper(),
-                        short_description=vuln.get("summary", ""),
-                        description=description,
-                        references=build_references(vuln.get("references", [])),
-                        security_issue=True,
-                        privacy_issue=False,
-                        has_public_exploit=False,
-                        targeted_by_malware=False,
-                        targeted_by_ransomware=False,
-                        targeted_by_nation_state=False,
-                        recommendation=recommendation,
-                    ),
-                    technical_detail=f"{vuln.get('details')} \n#### CVEs:\n {cve_list_details}",
-                    risk_rating=RISK_RATING_MAPPING[risk_rating.upper()],
+            parsed_vulns.extend(
+                parse_vulnerabilities(
+                    output=package,
+                    api_key=api_key,
+                    file_type=file_type,
+                    file_name=file_name,
                 )
+            )
+
+    return parsed_vulns
+
+
+def parse_vulnerabilities(
+    output: dict[str, Any],
+    package_name: str | None = None,
+    package_version: str | None = None,
+    api_key: str | None = None,
+    file_type: str | None = None,
+    file_name: str | None = None,
+) -> list[VulnData]:
+    """Parse the OSV API response to extract vulnerabilities.
+    Args:
+        output: The API response json.
+        package_name: The package name.
+        package_version: The package version.
+        api_key: The NVD API key.
+    Returns:
+        Parsed output.
+    """
+    try:
+        vulnerabilities = output.get("vulns") or output.get("vulnerabilities") or []
+        package_name = output.get("package", {}).get("name") or package_name
+        package_version = output.get("package", {}).get("version") or package_version
```

Comment:
Why would you get `package_version` if you have it in the params?

Reply:
sometimes I received it from the message and other times from the OSV output. It's a tradeoff to have it all in one function.